examples/webgl-vector-layer.js

```diff
@@ -0,0 +1,182 @@
+          tangentPx = normalize(tangentPx);
+          vec2 normalPx = vec2(-tangentPx.y, tangentPx.x);
+          float normalDir = vertexNumber < 0.5 || (vertexNumber > 1.5 && vertexNumber < 2.5) ? 1.0 : -1.0;
+          float tangentDir = vertexNumber < 1.5 ? 1.0 : -1.0;
+          float angle = vertexNumber < 1.5 ? v_angleStart : v_angleEnd;
+          vec2 offsetPx = getOffsetDirection(normalPx * normalDir, tangentDir * tangentPx, angle) * lineWidth * 0.5;
+          vec2 position =  vertexNumber < 1.5 ? a_segmentStart : a_segmentEnd;
+          gl_Position = u_projectionMatrix * vec4(position, 0.0, 1.0) + pxToScreen(offsetPx);
+          v_segmentStart = worldToPx(a_segmentStart);
+          v_segmentEnd = worldToPx(a_segmentEnd);
+
+          v_color = vec3(
+            floor(a_color / 256.0 / 256.0) / 256.0,
+            fract(floor(a_color / 256.0) / 256.0),
+            fract(a_color / 256.0)
+          );
+        }`,
+      lineStringFragmentShader: `
+        precision mediump float;
+        varying vec2 v_segmentStart;
+        varying vec2 v_segmentEnd;
+        varying float v_angleStart;
+        varying float v_angleEnd;
+
+        varying vec3 v_color;
+
+        float segmentDistanceField(vec2 point, vec2 start, vec2 end, float radius) {
+          vec2 startToPoint = point - start;
+          vec2 startToEnd = end - start;
+          float ratio = clamp(dot(startToPoint, startToEnd) / dot(startToEnd, startToEnd), 0.0, 1.0);
+          float dist = length(startToPoint - ratio * startToEnd);
+          return 1.0 - smoothstep(radius - 1.0, radius, dist);
+        }
+
+        float lineWidth = 1.5;
+
+        void main(void) {
+          gl_FragColor = vec4(v_color.rgb * 0.75, 1.0);
+          gl_FragColor *= segmentDistanceField(gl_FragCoord.xy, v_segmentStart, v_segmentEnd, lineWidth);
+        }`,
+      pointVertexShader: `
+        precision mediump float;
+        uniform mat4 u_projectionMatrix;
+        uniform mat4 u_offsetScaleMatrix;
+        attribute vec2 a_position;
+        attribute float a_index;
+        varying vec2 v_texCoord;
+
+        void main(void) {
+          mat4 offsetMatrix = u_offsetScaleMatrix;
+          float size = 6.0;
+          float offsetX = a_index == 0.0 || a_index == 3.0 ? -size / 2.0 : size / 2.0;
+          float offsetY = a_index == 0.0 || a_index == 1.0 ? -size / 2.0 : size / 2.0;
+          vec4 offsets = offsetMatrix * vec4(offsetX, offsetY, 0.0, 0.0);
+          gl_Position = u_projectionMatrix * vec4(a_position, 0.0, 1.0) + offsets;
+          float u = a_index == 0.0 || a_index == 3.0 ? 0.0 : 1.0;
+          float v = a_index == 0.0 || a_index == 1.0 ? 0.0 : 1.0;
+          v_texCoord = vec2(u, v);
+        }`,
+      pointFragmentShader: `
+        precision mediump float;
+        varying vec2 v_texCoord;
+
+        void main(void) {
+            gl_FragColor = vec4(1.0, 1.0, 1.0, 1.0);
+        }`,
```

Comment:
What do you think about having the shaders above be the defaults?  And we could have dark gray `color` attribute by default below.  Then people could try out the experimental renderer without having to copy/paste the shader code from this example.  And then they could experiment with their own color attribute as a first modification.

This feels like it would be a gentler introduction to the new functionality.  And since it is all experimental, we could still change things as we work on the higher level layer styling API.

Reply:
That would make sense; maybe having a `width` parameter as well for the lines.

Reply:
Done in the latest commits, there is now a separate module providing default shaders expecting attributes such as color, opacity and width. This has greatly simplified the example.

Hopefully the process of providing custom shaders isn't too obscure and error-prone, and relying on default ones should let people try this renderer without too much trouble!